src/cfnlint/rules/__init__.py

```diff
@@ -165,41 +199,16 @@ def __len__(self):
     def extend(self, more):
         """Extend rules"""
         for rule in more:
-            self.all_rules.append(rule)
-            if self.is_rule_enabled(rule.id, rule.experimental):
-                self.rules.append(rule)
-                if rule.id in self.configure_rules:
-                    rule.configure(self.configure_rules[rule.id])
+            self.register(rule)
 
     def __repr__(self):
         return '\n'.join([rule.verbose()
                           for rule in sorted(self.rules, key=lambda x: x.id)])
 
-    def is_rule_enabled(self, rule_id, experimental):
+    def is_rule_enabled(self, rule):
         """ Checks if an individual rule is valid """
-        # Evaluate experimental rules
-        if experimental and not self.include_experimental:
-            return False
-
-        # Evaluate includes first:
-        include_filter = False
-        for include_rule in self.include_rules:
-            if rule_id.startswith(include_rule):
-                include_filter = True
-        if not include_filter:
-            return False
-
-        # Enable mandatory rules without checking for if they are ignored
-        for mandatory_rule in self.mandatory_rules:
-            if rule_id.startswith(mandatory_rule):
-                return True
-
-        # Allowing ignoring of rules based on prefix to ignore checks
-        for ignore_rule in self.ignore_rules:
-            if rule_id.startswith(ignore_rule) and ignore_rule:
-                return False
-
-        return True
+        return rule.is_enabled(self.include_experimental, self.ignore_rules,
+                               self.include_rules, self.mandatory_rules)
```

Comment:
I like the switch to only passing `rule` into `is_rule_enabled()`

the new extra `rule.is_enabled()` layer confuses me a bit though, seems potentially unnecessary?

Reply:
I had to move the logic if a rule is enabled to the rule itself.  This is because E000X rules aren't part of the rule collection.  So I needed the same logic for checking if a rule is enabled if the rule belonged to a collection or not.